src/BenchmarkDotNet/Columns/MetricColumn.cs

```diff
@@ -23,13 +23,28 @@ public class MetricColumn : IColumn
 
         public bool IsDefault(Summary summary, BenchmarkCase benchmarkCase) => false;
 
-        public bool IsAvailable(Summary summary) => summary.Reports.Any(report => report.Metrics.ContainsKey(descriptor.Id));
+        public bool IsAvailable(Summary summary) => descriptor is Diagnosers.AllocatedMemoryMetricDescriptor || summary.Reports.Any(IsAvailable);
+
+        private bool IsAvailable(BenchmarkReport report)
+        {
+            if (!report.Metrics.TryGetValue(descriptor.Id, out var metric))
+            {
+                return false;
+            }
+            if (metric.Descriptor is Diagnosers.MemoryDiagnoser.GarbageCollectionsMetricDescriptor)
+            {
+                return metric.Value > 0;
+            }
+            return true;
+        }
 
         public string GetValue(Summary summary, BenchmarkCase benchmarkCase) => GetValue(summary, benchmarkCase, SummaryStyle.Default);
 
         public string GetValue(Summary summary, BenchmarkCase benchmarkCase, SummaryStyle style)
         {
-            if (!summary.HasReport(benchmarkCase) || !summary[benchmarkCase].Metrics.TryGetValue(descriptor.Id, out Metric metric) || (metric.Value == 0.0 && !style.PrintZeroValuesInContent))
+            if (!summary.HasReport(benchmarkCase) || !summary[benchmarkCase].Metrics.TryGetValue(descriptor.Id, out Metric metric))
+                return "?";
```

Comment:
Since `"?"` is reused in multiple places (e.g., e.g., `TryParse` from `GcStats`; `CsvMeasurementsExporter`), it will be better to extract it to a named constant so that we can easily change it in the future.